src/OrchardCore/OrchardCore/Localization/NullHtmlLocalizerFactory.cs

```diff
@@ -43,17 +42,26 @@ public IEnumerable<LocalizedString> GetAllStrings(bool includeParentCultures)
                         Array.Copy(pluralArgument.Arguments, 0, arguments, 1, pluralArgument.Arguments.Length);
                     }
 
-                    return new LocalizedHtmlString(name, translation, false, arguments);
+                    return ToHtmlString(StringLocalizer[name], translation, arguments);
                 }
             }
 
-            public LocalizedString GetString(string name) =>
-                NullStringLocalizerFactory.NullLocalizer.Instance.GetString(name);
+            public LocalizedString GetString(string name) => StringLocalizer.GetString(name);
 
-            public LocalizedString GetString(string name, params object[] arguments) =>
-                NullStringLocalizerFactory.NullLocalizer.Instance.GetString(name, arguments);
+            public LocalizedString GetString(string name, params object[] arguments)
+                => StringLocalizer.GetString(name, arguments);
 
-            IHtmlLocalizer IHtmlLocalizer.WithCulture(CultureInfo culture) => Instance;
+            IHtmlLocalizer IHtmlLocalizer.WithCulture(CultureInfo culture)
+                => new NullHtmlLocalizer();
+
+            private static IStringLocalizer StringLocalizer
+                => NullStringLocalizerFactory.NullStringLocalizer.Instance;
+
+            private LocalizedHtmlString ToHtmlString(LocalizedString localizedString)
```

Comment:
This is the only change I don't like. This method has no added value, it just makes the code more complex.

Reply:
It just uses the `LocalizedString` behind the scene, I saw similar approach in `HtmlStringLocalizer`, if you don't like it we can use `LocalizedString` instance directly in the return